sagemaker/pipeline/pipeline_parameters.py

```diff
@@ -0,0 +1,764 @@
+    cls : type
+        The dataclass type to instantiate
+    data : dict
+        The dictionary used to initialize the dataclass object
+
+    Returns
+    -------
+    object
+        An instance of the dataclass
+    """
+    if not is_dataclass(cls):
+        return data
+    field_types = {f.name: f.type for f in fields(cls)}
+    return cls(
+        **{
+            k: dict_to_dataclass(field_types[k], v)
+            for k, v in data.items()
+            if k in field_types
+        }
+    )
+
+
+def load_pipeline_args(filepath: str) -> PipelineArgs:
+    """Create PipelineArgs object from a JSON file.
+
+    Parameters
+    ----------
+    filepath : str
+        Path to the JSON representation of a PipelineArgs
+        instance.
+
+    Returns
+    -------
+    PipelineArgs
+        PipelineArgs instance with the loaded configuration
+    """
+    with open(filepath, "r", encoding="utf-8") as f:
+        config_dict = json.load(f)
+
+    args_instance = dict_to_dataclass(PipelineArgs, config_dict)
+
+    # Assertion to make the type checker happy
+    assert isinstance(args_instance, PipelineArgs)
+
+    return args_instance
+
+
+def parse_pipeline_args() -> PipelineArgs:
+    """Parses all the arguments for the pipeline definition.
+
+    Returns
+    -------
+    PipelineArgs
+        Pipeline configuration object.
+    """
+    parser = argparse.ArgumentParser(description="Create GraphStorm SageMaker Pipeline")
+
+    required_args = parser.add_argument_group("Required arguments")
+    optional_args = parser.add_argument_group("Optional arguments")
+
+    # AWS Configuration
+    required_args.add_argument(
+        "--execution-role",
+        type=str,
+        required=True,
+        help="SageMaker execution IAM role ARN. Required.",
+    )
+    required_args.add_argument(
+        "--region", type=str, required=True, help="AWS region. Required"
+    )
+    required_args.add_argument(
+        "--graphstorm-pytorch-cpu-image-url",
+        type=str,
+        required=True,
+        help="GraphStorm GConstruct/dist_part/train/inference CPU ECR image URL. Required",
+    )
+    optional_args.add_argument(
+        "--graphstorm-pytorch-gpu-image-url",
+        type=str,
+        help="GraphStorm GConstruct/dist_part/train/inference GPU ECR image URL.",
+    )
+    optional_args.add_argument(
+        "--gsprocessing-pyspark-image-url",
+        type=str,
+        help="GSProcessing SageMaker PySpark ECR image URL. Required if running GSProcessing",
+    )
+
+    # Instance Configuration
+    required_args.add_argument(
+        "--instance-count",
+        "--num-parts",
+        type=int,
+        help="Number of worker instances/partitions for partition, training, inference.",
+        required=True,
+    )
+    optional_args.add_argument(
+        "--cpu-instance-type",
+        type=str,
+        help=(
+            "CPU instance type. "
+            "Always used in DistPart step and if '--train-on-cpu' is provided, "
```

Comment:
What about gconstruct?

Reply:
That one uses `"--graph-construction-instance-type`